lib/code_corps/github/sync/issue/github_issue/github_issue.ex

```diff
@@ -17,93 +17,38 @@ defmodule CodeCorps.GitHub.Sync.Issue.GithubIssue do
   }
 
   alias Ecto.Changeset
-  alias Sync.User.GithubUser, as: GithubUserSyncer
 
   @typep linking_result :: {:ok, GithubIssue.t} | {:error, Changeset.t}
 
   @doc ~S"""
-  Finds or creates a `CodeCorps.GithubIssue` using the data in a GitHub Issue
-  payload.
+  Creates or updates a `CodeCorps.GithubIssue` from a github issue API payload.
 
-  The process is as follows:
-  - Search for the issue in our database with the payload data.
-   - If found, update it with payload data
-   - If not found, create it from payload data
+  The created record is associated to the provided `CodeCorps.GithubRepo` and, optionaly,
+  to a provided `CodeCorps.GithubPullRequest`.
 
-  `CodeCorps.GitHub.AdaptersIssue.to_issue/1` is used to adapt the payload data.
+  The created record is also associated with a matched `CodeCorps.GithubUser`, which is
+  created if necessary.
   """
-  @spec create_or_update_issue({GithubRepo.t, GithubPullRequest.t | nil}, map) :: linking_result
-  def create_or_update_issue({github_repo, github_pull_request}, %{"id" => github_issue_id} = attrs) do
-    params = to_params(attrs, github_repo, github_pull_request)
-    case Repo.get_by(GithubIssue, github_id: github_issue_id) do
-      nil -> create_issue(params)
-      %GithubIssue{} = issue -> update_issue(issue, params)
-    end
-  end
-
-  @doc ~S"""
-  Creates a `CodeCorps.GithubIssue` for the provided `CodeCorps.GithubRepo`
-  using specified attributes.
-
-  Links to existing `CodeCorps.GithubPullRequest` if matched by
-  `github_repo_id` and `number`.
-  """
-  @spec create_or_update_issue(GithubRepo.t, map) :: linking_result
-  def create_or_update_issue(%GithubRepo{} = repo, attrs) do
-    with {:ok, %GithubUser{} = github_user} <- GithubUserSyncer.create_or_update_github_user(attrs),
-         {:ok, %GithubIssue{} = github_issue} <- do_create_or_update_issue(repo, attrs, github_user)
-    do
-      {:ok, github_issue}
+  @spec create_or_update_issue(map, GithubRepo.t, GithubPullRequest.t | nil) :: linking_result
+  def create_or_update_issue(%{} = payload, %GithubRepo{} = github_repo, github_pull_request \\ nil) do
+    with {:ok, %GithubUser{} = github_user} <- Sync.User.GithubUser.create_or_update_github_user(payload) do
```

Comment:
Since this whole module is wrapped into a transaction, if this with fails, the transaction will still fail.

However, I do believe we should pull this `GithubUser` creation step out of this module and move it into the parent module, as an extra step of that multi.

The problem with that is that we would then also need an additional step in the github repo sync process -> `syncing github users`.

If we do something similar for comments, we would likely do `syncing github users for issues` and `syncing github users for comments`.

Or we could just not report those steps and consider them parts of `syncing issues` and `syncing comments` steps.

Either way, eliminating this `with` and moving it into `Sync.Issue` would flatten out the behavior and make it slightly more explicit.